src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1363,6 +1376,18 @@ def _handle_addons_args(cmd, addons_str, subscription_id, resource_group_name, a
     if 'azure-policy' in addons:
         addon_profiles['azurepolicy'] = ManagedClusterAddonProfile(enabled=True)
         addons.remove('azure-policy')
+    if 'ingress-appgw' in addons:
+        addon_profiles['IngressApplicationGateway'] = ManagedClusterAddonProfile(
+            enabled=True,
+            config={
+                'ApplicationGatewayName': appgw_name,
+                'SubnetPrefix': subnet_prefix,
+                'ApplicationGatewayId': appgw_id,
+                'SubnetId': subnet_id,
+                'Shared': appgw_shared,
```

Comment:
backend will handle the None case, right?

Reply:
yes, backend will handles None.